### DISCUSSION TITLE
Support Generic OIDC Providers

### DISCUSSION THREAD

**Discussion by acelinkio** [apache/airflow#39126] [General] [answered]
### Description

Follow up to #35131/#35237 which updated the docs to list which OIDC Providers are supported in the code.  This request is to add support for generic oidc providers.  There are a number of smaller OIDC Providers that would be awesome to allow authentication through.  Authentik, Discord, Authelia, AWS Cognito, and the growing number of alternatives.

As the previous ticket points out, the current implementation is limited to the case statement in get_oauth_user_info function located here: https://github.com/apache/airflow/blob/2.9.0/airflow/providers/fab/auth_manager/security_manager/override.py#L2169-L2276

It would be awesome to have that extended to cover additional providers.

### Use case/motivation

Would like to use Authenik to access AirTables

### Related issues

#35131/#35237

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment, marked as answer]
As explained in the other tickets - this is a request to Flask Application Builder - not to Airlfow. Airflow uses Flask Application Builder to implement those https://github.com/dpgaspar/Flask-AppBuilder . irflow is periodically syncing with what gets implemented in FAB - so I think you can open your issue there. 

It might be in the future we will have a generic AuthManager (Keycloak based) replacing FAB Auth Manager as we have today. The interface is already implemented - Airflow is looking for a contibution of an implementation (possibly based on KeyCloak  https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-56+Extensible+user+management) 

Conveting to a Discussion if more is needed.

**Reply by acelinkio** [reply]
Marking this as solved

I missed that piece in the previous conversation.  Thanks for the clarification and the additional context with how the internals work.  Really appreciate the links to the decision records.
